packages/shared/src/components/squads/SquadJoinButton.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { ReactElement, useMemo } from 'react';
+import classNames from 'classnames';
+import { useMutation, useQueryClient } from 'react-query';
+import { Squad } from '../../graphql/sources';
+import { Button } from '../buttons/Button';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { useToastNotification } from '../../hooks/useToastNotification';
+import { useLeaveSquad, useJoinSquad } from '../../hooks';
+import { labels } from '../../lib';
+
+type SquadJoinProps = {
+  className?: string;
+  squad: Squad;
+  joinText?: string;
+  leaveText?: string;
+};
+
+export const SquadJoinButton = ({
+  className,
+  squad,
+  joinText = 'Join squad',
+  leaveText = 'Leave squad',
+}: SquadJoinProps): ReactElement => {
+  const queryClient = useQueryClient();
+  const { displayToast } = useToastNotification();
+  const { squads, user, showLogin } = useAuthContext();
+  const isCurrentMember = useMemo(
+    () => squads?.some(({ id }) => id === squad.id) || false,
+    [squad, squads],
+  );
```

Comment:
Does squad not have a `currentMember` property already?

Reply:
It was because squad query would get stale due to only depending on handle and `currentMember` would be left false. Fixed here 06aa0932e015cb18fbc12d3ed37ce8f7625fe01a